examples/erc1155/tests/erc1155.rs

```diff
@@ -2173,3 +2177,303 @@ async fn support_interface(alice: Account) -> eyre::Result<()> {
 
     Ok(())
 }
+
+// ============================================================================
+// Integration Tests: ERC-1155 Pausable Extension
+// ============================================================================
+
+#[e2e::test]
+async fn pauses(alice: Account) -> eyre::Result<()> {
+    let contract_addr = alice
+        .as_deployer()
+        .with_default_constructor::<constructorCall>()
+        .deploy()
+        .await?
+        .address()?;
+
+    let contract = Erc1155::new(contract_addr, &alice.wallet);
+
+    let receipt = receipt!(contract.pause())?;
+
+    assert!(receipt.emits(Erc1155::Paused { account: alice.address() }));
+
+    let paused = contract.paused().call().await?.paused;
+
+    assert!(paused);
+
+    Ok(())
+}
+
+#[e2e::test]
+async fn unpauses(alice: Account) -> eyre::Result<()> {
+    let contract_addr = alice
+        .as_deployer()
+        .with_default_constructor::<constructorCall>()
+        .deploy()
+        .await?
+        .address()?;
+
+    let contract = Erc1155::new(contract_addr, &alice.wallet);
+
+    let _ = watch!(contract.pause())?;
+
+    let receipt = receipt!(contract.unpause())?;
+
+    assert!(receipt.emits(Erc1155::Unpaused { account: alice.address() }));
+
+    let paused = contract.paused().call().await?.paused;
+
+    assert_eq!(false, paused);
+
+    Ok(())
+}
+
+#[e2e::test]
+async fn mint_reverts_in_paused_state(alice: Account) -> eyre::Result<()> {
+    let contract_addr = alice
+        .as_deployer()
+        .with_default_constructor::<constructorCall>()
+        .deploy()
+        .await?
+        .address()?;
+    let contract = Erc1155::new(contract_addr, &alice.wallet);
+
+    let alice_addr = alice.address();
+    let token_id = random_token_ids(1)[0];
+    let value = random_values(1)[0];
+
+    let _ = watch!(contract.pause())?;
+
+    let err = send!(contract.mint(
+        alice_addr,
+        token_id,
+        value,
+        vec![0, 1, 2, 3].into()
+    ))
+    .expect_err("should return EnforcedPause");
+
+    assert!(err.reverted_with(Erc1155::EnforcedPause {}));
+
+    Ok(())
+}
+
+#[e2e::test]
+async fn mint_batch_reverts_in_paused_state(
+    alice: Account,
+    bob: Account,
+    dave: Account,
+) -> eyre::Result<()> {
+    let contract_addr = alice
+        .as_deployer()
+        .with_default_constructor::<constructorCall>()
+        .deploy()
+        .await?
+        .address()?;
+    let contract = Erc1155::new(contract_addr, &alice.wallet);
+
+    let alice_addr = alice.address();
+    let bob_addr = bob.address();
+    let dave_addr = dave.address();
+    let token_ids = random_token_ids(3);
+    let values = random_values(3);
+
+    let accounts = vec![alice_addr, bob_addr, dave_addr];
```

Comment:
What is the benefit of testing `mintBatch` on each of these accounts? I think testing against just one would be sufficient

Reply:
Not needed, good catch!